sql/core/src/test/scala/org/apache/spark/sql/sources/DisableUnnecessaryBucketedScanSuite.scala

```diff
@@ -218,4 +221,27 @@ abstract class DisableUnnecessaryBucketedScanSuite extends QueryTest with SQLTes
       }
     }
   }
+
+  test("SPARK-33075: not disable bucketed table scan for cached query") {
+    withTable("t1") {
+      withSQLConf(SQLConf.AUTO_BUCKETED_SCAN_ENABLED.key -> "true") {
+        df1.write.format("parquet").bucketBy(8, "i").saveAsTable("t1")
+        sql("CACHE TABLE tempTable AS SELECT i FROM t1")
```

Comment:
why not just `CACHE TABLE t1`?

Reply:
Either way is fine for me, if you think it's too redundant I can also change that.

Reply:
yea let's be simpler.

Reply:
it can also save the `uncache` at the end, as the table will be dropped at the end.

Reply:
@cloud-fan - sure, updated.